src/content/blog/astro-510.mdx

```diff
@@ -0,0 +1,146 @@
+```
+The same session data is available in your Astro Actions:
+
+```ts {9,11}
+// src/actions/addToCart.ts
+import { defineAction } from "astro:actions";
+import { z } from "astro:schema";
+
+export const server = {
+  addToCart: defineAction({
+    input: z.object({ productId: z.string() }),
+    handler: async (input, context) => {
+      const cart = await context.session.get("cart");
+      cart.push(input.productId);
+      await context.session.set("cart", cart);
+      return cart;
+    },
+  }),
+};
+```
+
+Sessions work on pages that are rendered on-demand, or they pair perfectly with [server islands](https://docs.astro.build/en/concepts/islands/#server-islands), letting you sprinkle islands of personalized content in an otherwise static site.
+
+```astro
+---
+import CartButton from '../components/CartButton.astro';
+---
+<CartButton server:defer />
+```
+
+### Drivers
+
+Astro sessions are powered by [unstorage](https://unstorage.unjs.io/) drivers, which are modules that Astro uses to store the session data. Unstorage supports dozens of providers, giving you a world of choice for storing the user data. When sessions are a stable feature, adapters will automatically configure a default driver, but during the experimental phase you must choose and configure one yourself. When using the Node.js adapter, the best choice is usually the [filesystem driver](https://unstorage.unjs.io/drivers/fs), but this cannot be used in serverless environments as they have no shared filesystem. For these, the best default is usually the storage feature provided by the platform, such as the [Netlify Blobs driver](https://unstorage.unjs.io/drivers/netlify), the [Cloudflare KV driver](https://unstorage.unjs.io/drivers/cloudflare) or the [Deno KV driver](https://unstorage.unjs.io/drivers/deno). You can also use a cross-platform driver such as [Upstash](https://unstorage.unjs.io/drivers/upstash) or [Redis](https://unstorage.unjs.io/drivers/redis).
+
+### Usage
+
+To try experimental sessions, add a `sessions` key to your `astro.config.mjs`:
+
+This example shows how to enable the filesystem driver, which works with the Node.js adapter:
+
+```js title="astro.config.mjs" ins={4-8}
+  {
+    adapter: node({ mode: 'standalone' }),
+    experimental: {
+      session: {
+        // Required: the name of the Unstorage driver
+        driver: "fs",
+      },
+    },
+  }
+```
+
+For more information see [the docs for the experimental flag](https://docs.astro.build/en/reference/experimental-flags/sessions/). We'd also appreciate your feedback on the feature. For more details, see [the Sessions RFC](https://github.com/withastro/roadmap/pull/1055) and leave your comments.
+
+## Remote image caching
+
+Astro sites can use the built-in `<Image />` component to [display optimized images](https://docs.astro.build/en/guides/images/#display-optimized-images-with-the-image--component) with automatic resizing and optimization. This works with both local and remote images. 
```

Comment:
```suggestion
You already know and love the built-in `<Image />` component to [display optimized images](https://docs.astro.build/en/guides/images/#display-optimized-images-with-the-image--component) with automatic resizing and optimization. While this component currently transforms both local and remote images, image caching between builds was limited to local images. 
```
Getting the history out of the way sooner, and emphasizing what's old vs what's new with a tiny bit of framing, and fitting everything up until the "NOW...." part in this first paragraph.